lib/rubocop/cop/style/negated_if_else_condition.rb

```diff
@@ -98,21 +97,38 @@ def swap_branches(corrector, node)
           if node.if_branch.nil?
             corrector.remove(range_by_whole_lines(node.loc.else, include_final_newline: true))
           else
-            if_range = node_with_comments(node.if_branch)
-            else_range = node_with_comments(node.else_branch)
+            if_range = if_range(node)
+            else_range = else_range(node)
 
             corrector.replace(if_range, else_range.source)
             corrector.replace(else_range, if_range.source)
           end
         end
 
-        def node_with_comments(node)
-          first_statement = node.begin_type? ? node.children[0] : node
-          return node if processed_source.ast_with_comments[first_statement].empty?
+        # Collect the entire if branch, including whitespace and comments
+        def if_range(node)
+          if node.ternary?
+            node.if_branch
+          else
+            Parser::Source::Range.new(
+              processed_source.buffer,
+              node.condition.loc.expression.end_pos,
+              node.loc.else.begin_pos
+            )
```

Comment:
That's just my two cents :-)

```suggestion
            range_between(node.condition.loc.expression.end_pos, node.loc.else.begin_pos)
```